src/tmiic.cpp

```diff
@@ -0,0 +1,200 @@
+//-----------------------------------------------------------------------------
+void repeatEdgesOverHistory (Environment& environment) {
+  //
+  // Now, we iterate over computed edges to duplicate (if needed)
+  // the edges over the history
+  //
+  int n_nodes_not_lagged = environment.n_nodes / (environment.tau + 1);
+  auto& edge_list = environment.connected_list;
+  auto edge_end = end(edge_list);
+  for (auto iter0 = begin(edge_list); iter0 != edge_end; ++iter0) {
+    //
+    // Find the Edge structure that will be duplicated 
+    //
+    int node1_pos = iter0->X;
+    int node2_pos = iter0->Y;
+    const Edge& edge_orig = environment.edges(node1_pos, node2_pos);
+    //
+    // To create the same edge over history, we switch the nodes pos by n_nodes_not_lagged
+    // until node pos > nb total nodes
+    //
+    while (true) {
+      node1_pos += n_nodes_not_lagged;
+      node2_pos += n_nodes_not_lagged;
+      if ( (node1_pos >= environment.n_nodes) || (node2_pos >= environment.n_nodes) )
+        break;
+      //
+      // Duplicate the edge info into environment.edges array
+      //
+      Edge& edge_to_modif = environment.edges(node1_pos,node2_pos);
+      edge_to_modif.status = edge_orig.status;
+      edge_to_modif.status_init = edge_orig.status_init;
+      edge_to_modif.status_prev = edge_orig.status_prev;
+
+      Edge& edge_to_modif_inverse = environment.edges(node2_pos,node1_pos);
+      edge_to_modif_inverse.status = edge_orig.status;
+      edge_to_modif_inverse.status_init = edge_orig.status_init;
+      edge_to_modif_inverse.status_prev = edge_orig.status_prev;
+      //
+      // Add the nodes into connected_list
+      //
+      environment.connected_list.emplace_back (node1_pos, node2_pos, environment.edges(node1_pos,node2_pos) );
+    }
+  }
+}
+
+//-----------------------------------------------------------------------------
+// completeOrientationUsingTime
+//-----------------------------------------------------------------------------
+// Description: Complete the orientations with the orientation of temporal  
+// edges that were not previously oriented 
+//
+// Detail: completeOrientationUsingTime will look in the list of connected
+// edges the ones that have not been oriented using the unshielded triples, 
+// are lagged and have a lag0 node (edges not having a lag0 are only a 
+// consequence of duplicating edges over history to maximize the number of 
+// unshielded triples and we are not interested to orient them as they will
+// be removed at the end of the orientation step). Edges matching these
+// criteria will be oriented using time from oldest node to newest node.
+//
+// Params:
+// - Environment&: the environment structure
+// - std::vector<Triple>& : list of unshielded triples
+//--------------------------------------------------------------------------------
+void completeOrientationUsingTime (Environment& environment, 
+                                   const std::vector<Triple>& triples) {
```

Comment:
As described in your comments above, this function finds specific triples and applies orientations, so I'd suggest (looks more natural to me) drop the `vector<Triple>` parameter, but instead create its own list inside the function, similar to what does the function `orientationProbability`.